pkg/cmd/cli/install/install.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2019 the Heptio Ark contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package install
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/spf13/pflag"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	api "github.com/heptio/velero/pkg/apis/velero/v1"
+	"github.com/heptio/velero/pkg/client"
+	"github.com/heptio/velero/pkg/cmd"
+	"github.com/heptio/velero/pkg/cmd/util/output"
+	"github.com/heptio/velero/pkg/install"
+)
+
+const defaultImage = "gcr.io/heptio/velero:latest"
+
+type InstallOptions struct {
+	Namespace                 string
+	DeploymentName            string
+	Image                     string
+	BucketName                string
+	Prefix                    string
+	BackupStorageProviderName string
+	RestoreOnly               bool
+	LogLevel                  string
+	ResticTimeout             time.Duration
+	SecretName                string
+}
+
+// Args:
+//   secret
+//   plugin-dir
+//   basically all the server args?
+// Flags
+//   dry-run (yaml dump)
+
+func (o *InstallOptions) BindFlags(flags *pflag.FlagSet) {
+	// TODO Send this string down into the deployment
+	flags.StringVar(&o.DeploymentName, "deploy-name", "", "name to apply to the Velero deployment")
+	flags.StringVar(&o.BucketName, "bucket-name", "", "bucket name in which to store backups")
+	flags.StringVar(&o.Prefix, "prefix", "", "prefix under the bucket in which to store backups")
+	flags.StringVar(&o.BackupStorageProviderName, "backup-provider", "", "provider name for backup storage")
+	flags.StringVar(&o.Image, "image", "", "image to use for the Velero server deployment")
+	flags.BoolVar(&o.RestoreOnly, "restore-only", false, "run the server in restore-only mode")
+}
+
+func NewInstallOptions() *InstallOptions {
+	return &InstallOptions{}
+}
+
+func NewCommand(f client.Factory) *cobra.Command {
+	o := NewInstallOptions()
+	c := &cobra.Command{
+		Use:   "install",
+		Short: "Install Velero",
+		Long:  "Install Velero",
+		Run: func(c *cobra.Command, args []string) {
+			o.Namespace = c.Flag("namespace").Value.String()
+			cmd.CheckError(o.Complete(args, f))
+			cmd.CheckError(o.Run(c))
+		},
+	}
+
+	o.BindFlags(c.Flags())
+	output.BindFlags(c.Flags())
+	output.ClearOutputFlagDefault(c)
+
+	return c
+}
+
+func (o *InstallOptions) Run(c *cobra.Command) error {
+
+	//TODO: Extract this logic into a function inside of the install package that anyone could use.
+	var resources []runtime.Object
+
+	crds := install.CRDs()
```

Comment:
Not suer I'm reading this right: `CRDs` is in this same package right, do we need the `install.`?

Reply:
Also, could we just do `range CRDs`?

Reply:
Probably could range.

`install` is `"github.com/heptio/velero/pkg/install"`, not this package. I'm not wild about that naming, but this package is following the same conventions as the rest of the client commands (package named after the command)